src/main/java/io/cdap/plugin/gcp/bigtable/source/BigtableSourceConfig.java

```diff
@@ -0,0 +1,238 @@
+    this.project = project;
+    this.serviceFilePath = serviceFilePath;
+    this.keyAlias = keyAlias;
+    this.scanRowStart = scanRowStart;
+    this.scanRowStop = scanRowStop;
+    this.scanTimeRangeStart = scanTimeRangeStart;
+    this.scanTimeRangeStop = scanTimeRangeStop;
+    this.onError = onError;
+    this.schema = schema;
+  }
+
+  public void validate() {
+    super.validate();
+    if (!containsMacro(PROJECT) && null == tryGetProject()) {
+      throw new InvalidConfigPropertyException("Could not detect Google Cloud project id from the environment. " +
+                                                 "Please specify a project id.", PROJECT);
+    }
+    if (!containsMacro(INSTANCE) && Strings.isNullOrEmpty(instance)) {
+      throw new InvalidConfigPropertyException("Instance ID must be specified", INSTANCE);
+    }
+    String serviceAccountFilePath = getServiceAccountFilePath();
+    if (!containsMacro(SERVICE_ACCOUNT_FILE_PATH) && null != serviceAccountFilePath) {
+      File serviceAccountFile = new File(serviceAccountFilePath);
+      if (!serviceAccountFile.exists()) {
+        throw new InvalidConfigPropertyException(
+          String.format("Service account file '%s' does not exist", serviceAccountFilePath),
+          SERVICE_ACCOUNT_FILE_PATH
+        );
+      }
+    }
+    if (!containsMacro(ON_ERROR) && null != onError) {
+      if (Strings.isNullOrEmpty(onError)) {
+        throw new InvalidConfigPropertyException("Error handling must be specified", ON_ERROR);
+      }
+      if (null == ErrorHandling.fromDisplayName(onError)) {
+        throw new InvalidConfigPropertyException("Invalid record error handling strategy name", ON_ERROR);
+      }
+    }
+    if (!containsMacro(SCHEMA)) {
+      Schema parsedSchema = getSchema();
+      if (null == parsedSchema) {
+        throw new InvalidConfigPropertyException("Schema must be specified", SCHEMA);
+      }
+      if (Schema.Type.RECORD != parsedSchema.getType()) {
+        String message = String.format("Schema is of invalid type '%s'. The schema must be a record.",
+                                       parsedSchema.getType());
+        throw new InvalidConfigPropertyException(message, SCHEMA);
+      }
+      List<Schema.Field> fields = parsedSchema.getFields();
+      if (null == fields || fields.isEmpty()) {
+        throw new InvalidConfigPropertyException("Schema should contain fields to map", SCHEMA);
+      }
+      for (Schema.Field field : fields) {
+        Schema.Type fieldType = field.getSchema().isNullable() ?
+          field.getSchema().getNonNullable().getType() : field.getSchema().getType();
+        if (!SUPPORTED_FIELD_TYPES.contains(fieldType)) {
+          String supportedTypes = SUPPORTED_FIELD_TYPES.stream()
+            .map(Enum::name)
+            .map(String::toLowerCase)
+            .collect(Collectors.joining(", "));
+          String errorMessage = String.format("Wrong type '%s' for field '%s'. Supported types are: '%s'",
```

Comment:
nit: Field '%s' is of unsupported type '%s'. Supported types are: %s

Reply:
fixed